### PR TITLE
changelog: update missing 8.17 release notes

### PR DESCRIPTION
**Pull request by 1pkg**
## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

### CODE REVIEW

**Review by endorama — APPROVED** [review]

File: changelogs/8.17.asciidoc [hunk 1]
```diff
@@ -25,5 +25,6 @@ https://github.com/elastic/apm-server/compare/v8.16.0\...v8.17.0[View commits]
 [float]
 ==== Added
 
+- Surface config parsing error under EA managed mode. {pull}14574[14574]
```
Comment (carsonip) [inline changelogs/8.17.asciidoc:28]:
```suggestion
- Surface config parsing error under EA managed mode by logging and marking EA input unit as failed {pull}14574[14574]
- Remove unnecessary hot reload under EA managed mode when apm tracing config is nil {pull}14865[14865]
```

Both of these should be moved to "Bug fixes" from "Added".

**Review by carsonip — COMMENTED** [review]
Thanks!

**Review by carsonip — COMMENTED** [review]
Have to admit I'm not familiar with major version changes. Can you check if the missing 8.17 changelog in main branch is intended? I've noticed it since the target branch of this PR is 8.x.

File: changelogs/8.17.asciidoc (hunk 1 above)
Comment (1pkg) [reply changelogs/8.17.asciidoc]: updated

**Comment by 1pkg** [comment]
@mergify backport 8.17

**Pull request merged**